noir-projects/noir-protocol-circuits/crates/types/src/transaction/tx_context.nr

```diff
@@ -1,61 +1,64 @@
 use crate::{
-    constants::{GENERATOR_INDEX__TX_CONTEXT, TX_CONTEXT_DATA_LENGTH}, hash::pedersen_hash,
-    traits::{Deserialize, Hash, Serialize, Empty}, utils::reader::Reader
+    constants::{GENERATOR_INDEX__TX_CONTEXT, TX_CONTEXT_LENGTH}, hash::pedersen_hash,
+    traits::{Deserialize, Hash, Serialize, Empty}, utils::reader::Reader,
+    abis::gas_settings::GasSettings
 };
 
+// docs:start:tx-context
 struct TxContext {
-    is_fee_payment_tx : bool,
-    is_rebate_payment_tx : bool,
-
     chain_id : Field,
     version : Field,
+    gas_settings: GasSettings,
```

Comment:
IIUC transactions that modify this (e.g. with a mutable reference to `context.inputs.tx_context.gas_settings = something_large`) can't change the tx fee because the private context's `finish()` does not pass the tx_context as output. 

The PXE is responsible for passing the same tx_context to the execution and to the kernel?

Reply:
Not sure I understand the question. Once TxContext is set from the TxExecutionRequest, it should not be changed. The kernel is responsible for asserting that the tx_context in the PublicCircuitPublicInputs matches the one on its ConstantData, which is set from the TxExecutionRequest. If the PXE doesn't play along, the kernel should fail.